libraries/AP_ADSB/AP_ADSB.cpp

```diff
@@ -214,11 +221,26 @@ void AP_ADSB::deinit(void)
         delete [] in_state.vehicle_list;
         in_state.vehicle_list = nullptr;
     }
+    for (uint8_t i=0; i<ADSB_MAX_INSTANCES; i++) {
+        if (_backend[i] != nullptr) {
+            delete _backend[i];
```

Comment:
I think we mark the TYPE as reboot required, and don't free up backends at runtime. Otherwise we're opening ourselves up to nasty race conditions, for example if mavlink processing from another thread, or if a lua script tries to access a backend